tests/mocha/field_angle_test.js

```diff
@@ -0,0 +1,234 @@
+    });
+    test('Non-Parsable String', function() {
+      var angleField = new Blockly.FieldAngle('bad');
+      assertValueDefault(angleField);
+    });
+    test('NaN', function() {
+      var angleField = new Blockly.FieldAngle(NaN);
+      assertValueDefault(angleField);
+    });
+    test('Integer', function() {
+      var angleField = new Blockly.FieldAngle(1);
+      assertValue(angleField, 1);
+    });
+    test('Float', function() {
+      var angleField = new Blockly.FieldAngle(1.5);
+      assertValue(angleField, 1.5);
+    });
+    test('Integer String', function() {
+      var angleField = new Blockly.FieldAngle('1');
+      assertValue(angleField, 1);
+    });
+    test('Float String', function() {
+      var angleField = new Blockly.FieldAngle('1.5');
+      assertValue(angleField, 1.5);
+    });
+  });
+  suite('fromJson', function() {
+    test('Empty', function() {
+      var angleField = Blockly.FieldAngle.fromJson({});
+      assertValueDefault(angleField);
+    });
+    test('Null', function() {
+      var angleField = Blockly.FieldAngle.fromJson({ angle:null });
+      assertValueDefault(angleField);
+    });
+    test('Undefined', function() {
+      var angleField = Blockly.FieldAngle.fromJson({ angle:undefined });
+      assertValueDefault(angleField);
+    });
+    test('Non-Parsable String', function() {
+      var angleField = Blockly.FieldAngle.fromJson({ angle:'bad' });
+      assertValueDefault(angleField);
+    });
+    test('NaN', function() {
+      var angleField = Blockly.FieldAngle.fromJson({ angle:NaN });
+      assertValueDefault(angleField);
+    });
+    test('Integer', function() {
+      var angleField = Blockly.FieldAngle.fromJson({ angle:1 });
+      assertValue(angleField, 1);
+    });
+    test('Float', function() {
+      var angleField = Blockly.FieldAngle.fromJson({ angle:1.5 });
+      assertValue(angleField, 1.5);
+    });
+    test('Integer String', function() {
+      var angleField = Blockly.FieldAngle.fromJson({ angle:'1' });
+      assertValue(angleField, 1);
+    });
+    test('Float String', function() {
+      var angleField = Blockly.FieldAngle.fromJson({ angle:'1.5' });
+      assertValue(angleField, 1.5);
+    });
+  });
+  suite('setValue', function() {
+    test('Empty -> Null', function() {
+      var angleField = new Blockly.FieldAngle();
+      angleField.setValue(null);
+      assertValueDefault(angleField);
+    });
+    test('Value -> Null', function() {
+      var angleField = new Blockly.FieldAngle(1);
+      angleField.setValue(null);
+      assertValue(angleField, 1);
+    });
+    test('Empty -> Undefined', function() {
+      var angleField = new Blockly.FieldAngle();
+      angleField.setValue(undefined);
+      assertValueDefault(angleField);
+    });
+    test('Value -> Undefined', function() {
+      var angleField = new Blockly.FieldAngle(1);
+      angleField.setValue(undefined);
+      // TODO: Is this how we want this to work?
```

Comment:
I think this is a bug in field angle. Should probably be treated the same as null.